pennylane/transforms/condition.py

```diff
@@ -88,21 +87,148 @@ def cond(condition, true_fn, false_fn=None):
+
+    .. code-block :: pycon
+
+        >>> first_par = np.array(0.3, requires_grad=True)
+        >>> sec_par = np.array(1.23, requires_grad=True)
+        >>> qnode(first_par, sec_par)
+        tensor(0.32677361, requires_grad=True)
+
+    .. note::
+
+        If the first argument of ``cond`` is a measurement value (e.g., ``m_0``
+        in ``qml.cond(m_0, qml.RY)``), then ``m_0 == 1`` is considered
+        internally.
+
+    .. UsageDetails::
+
+        **Conditional quantum functions**
+
+        The ``cond`` transform allows conditioning quantum functions too:
+
+        .. code-block:: python3
+
+            dev = qml.device("default.qubit", wires=2)
+
+            def qfunc(par, wires):
+                qml.Hadamard(wires[0])
+                qml.RY(par, wires[0])
+
+            @qml.qnode(dev)
+            def qnode(x):
+                qml.Hadamard(0)
+                m_0 = qml.measure(0)
+                qml.cond(m_0, qfunc)(x, wires=[1])
+                return qml.expval(qml.PauliZ(1))
+
+        .. code-block :: pycon
+
+            >>> par = np.array(0.3, requires_grad=True)
+            >>> qnode(par)
+            tensor(0.3522399, requires_grad=True)
+
+        **Passing two quantum functions**
+
+        In the qubit model, single-qubit measurements may result in one of two
+        outcomes. Such measurement outcomes may then be used to create
+        conditional expressions.
+
+        According to the truth value of the conditional expression passed to
+        ``cond``, the transform can apply a quantum function in both the
+        ``True`` and ``False`` case:
+
+        .. code-block:: python3
+
+            dev = qml.device("default.qubit", wires=2)
+
+            def qfunc1(x, wires):
+                qml.Hadamard(wires[0])
+                qml.RY(x, wires[0])
+
+            def qfunc2(x, wires):
+                qml.Hadamard(wires[0])
+                qml.RZ(x, wires[0])
+
+            @qml.qnode(dev)
+            def qnode(x):
+                qml.Hadamard(0)
+                m_0 = qml.measure(0)
+                qml.cond(m_0, qfunc1, qfunc2)(x, wires=[1])
+                return qml.expval(qml.PauliZ(1))
+
+        .. code-block :: pycon
+
+            >>> par = np.array(0.3, requires_grad=True)
+            >>> qnode1(par)
+            tensor(-0.1477601, requires_grad=True)
+
+        The previous QNode is equivalent to using ``cond`` twice, inverting the
+        conditional expression in the second case using the ``~`` unary
+        operator:
+
+        .. code-block:: python3
+
+            @qml.qnode(dev)
+            def qnode2(x):
+                qml.Hadamard(0)
+                m_0 = qml.measure(0)
+                qml.cond(m_0, qfunc1)(x, wires=[1])
+                qml.cond(~m_0, qfunc2)(x, wires=[1])
+                return qml.expval(qml.PauliZ(1))
+
+        .. code-block :: pycon
+
+            >>> qnode2(par)
+            tensor(-0.1477601, requires_grad=True)

```

Comment:
```suggestion
            tensor(0.97766824, requires_grad=True)
```

Reply:
Note: this is different from the previous result on `v0.22.0-rc0`, but it's due to a bug with MV inversion. That bug is being resolved in this PR and it results in getting the same result as for `qnode1`. Reverted the result.